pkg/network/ebpf/c/ip.h

```diff
@@ -36,12 +36,18 @@ static __always_inline __u64 read_conn_tuple_skb(struct __sk_buff *skb, skb_info
     __u8 l4_proto = 0;
     switch (l3_proto) {
     case ETH_P_IP:
+    {
+        __u16 ipv4_hdr_len = (load_byte(skb, info->data_off) & 0x0f) << 2;
```

Comment:
This depends on the endianness of the host. For a LE host, I believe this should be `0xf0` instead, based on looking at the `iphdr` definition. https://github.com/torvalds/linux/blob/d0ee23f9d78be5531c4b055ea424ed0b489dfe9b/include/uapi/linux/ip.h#L86-L93

Reply:
This is true for for bit fields as order matter.
But not for (u8) load_byte() there are no LE/BE byte

Reply:
I would update the `ipv4_hdr_len` type tu __u8 (may this confused you) as the maximum would be 60 bytes (15*4)

Reply:
Yeah, you are right. I got confused by the bitfield weirdness.